python/xorq/ibis_yaml/udf.py

```diff
@@ -0,0 +1,97 @@
+import base64
+from typing import Any
+
+import cloudpickle
+
+import xorq.vendor.ibis.expr.datatypes as dt
+import xorq.vendor.ibis.expr.operations as ops
+import xorq.vendor.ibis.expr.rules as rlz
+from xorq.ibis_yaml.common import (
+    _translate_type,
+    register_from_yaml_handler,
+    translate_from_yaml,
+    translate_to_yaml,
+)
+from xorq.ibis_yaml.utils import freeze
+from xorq.vendor.ibis.common.annotations import Argument
+
+
+def serialize_udf_function(fn: callable) -> str:
+    pickled = cloudpickle.dumps(fn)
+    encoded = base64.b64encode(pickled).decode("ascii")
+    return encoded
+
+
+def deserialize_udf_function(encoded_fn: str) -> callable:
+    pickled = base64.b64decode(encoded_fn)
+    return cloudpickle.loads(pickled)
+
+
+@translate_to_yaml.register(ops.ScalarUDF)
+def _scalar_udf_to_yaml(op: ops.ScalarUDF, compiler: Any) -> dict:
+    if getattr(op.__class__, "__input_type__", None) != ops.udf.InputType.BUILTIN:
+        raise NotImplementedError(
+            f"Translation of UDFs with input type {getattr(op.__class__, '__input_type__', None)} is not supported"
+        )
+    arg_names = [
+        name
+        for name in dir(op)
+        if not name.startswith("__") and name not in op.__class__.__slots__
+    ]
+
+    return freeze(
+        {
+            "op": "ScalarUDF",
+            "unique_name": op.__func_name__,
```

Comment:
is this name supposed to be unique in the sense of a hash?

Reply:
No, I don't think the name needs to be unique like a hash. I will update for clarity.